source/extensions/tracers/opentelemetry/http_trace_exporter.cc

```diff
@@ -0,0 +1,86 @@
+#include "http_trace_exporter.h"
+
+#include <chrono>
+#include <memory>
+#include <string>
+
+#include "source/common/common/logger.h"
+#include "source/common/protobuf/protobuf.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace OpenTelemetry {
+
+OpenTelemetryHttpTraceExporter::OpenTelemetryHttpTraceExporter(
+    Upstream::ClusterManager& cluster_manager,
+    envoy::config::trace::v3::OpenTelemetryConfig::HttpConfig http_config,
+    OpenTelemetryTracerStats& tracing_stats)
+    : cluster_manager_(cluster_manager), http_config_(http_config), tracing_stats_(tracing_stats) {}
+
+bool OpenTelemetryHttpTraceExporter::log(const ExportTraceServiceRequest& request) {
+
+  std::string request_body;
+
+  const auto ok = request.SerializeToString(&request_body);
+  if (!ok) {
+    ENVOY_LOG(warn, "Error while serializing the binary proto ExportTraceServiceRequest.");
+    return false;
+  }
+
+  Http::RequestMessagePtr message = std::make_unique<Http::RequestMessageImpl>();
+  message->headers().setReferenceMethod(Http::Headers::get().MethodValues.Post);
+  message->headers().setReferenceContentType(Http::Headers::get().ContentTypeValues.Protobuf);
+
+  // If traces_path is omitted, send to /v1/traces by default
+  if (http_config_.traces_path().empty()) {
+    message->headers().setPath(TRACES_PATH);
+  } else {
+    message->headers().setPath(http_config_.traces_path());
+  }
+
+  // TODO: Can we get the hostname that is configured in the cluster "socker_address" field?
+  message->headers().setHost(http_config_.hostname());
```

Comment:
@htuch @AlexanderEllis do you know why we need to set the host name header here? In my tests, if I don't set it and if it doesn't match to the address on the `cluster` definition, nothing works. Is my understanding right that Envoy uses the `host` header to "redirect" it internally to the proper cluster? 

If the above is correct, do you think it makes sense to try to get the host name value from the `cluster` configuration (`address` property)? It might be an IP so I'm not sure that makes sense always. But I find it "odd" that I have to define it twice in my config, such as:

```
tracing:
            provider:
              name: envoy.tracers.opentelemetry
              typed_config:
                service_name: envoy-HTTP-exporter
                http_config:
                  cluster_name: my-cluster
                  hostname: "my-cluster.com"
                  timeout: 10s
                  
# further down, cluster config
clusters:

- name: my-cluster
    type: LOGICAL_DNS
    dns_lookup_family: V4_ONLY
    lb_policy: ROUND_ROBIN
    load_assignment:
      cluster_name: my-cluster
      endpoints:
      - lb_endpoints:
        - endpoint:
            address:
              socket_address:
                address: "my-cluster.com"

```